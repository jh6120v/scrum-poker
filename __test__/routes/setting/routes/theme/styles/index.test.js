import 'jest-styled-components';
import { render } from '@testing-library/react';
import React from 'react';
import {
    ThemePreview, ThemePreviewCardBack, ThemePreviewCardBackInner,
    ThemePreviewCardFront, ThemePreviewCardFrontInner, ThemeSetting, ThemeSettingItem, ThemeSettingList,
    ThemeSettingPicker
} from '../../../../../../src/routes/setting/routes/theme/styles';

describe('test theme styles.', () => {
    it('test <ThemePreview /> snapshot.', () => {
        const { container } = render(<ThemePreview />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemePreview /> when render with default props.', () => {
        const { container } = render(<ThemePreview />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('border-radius', '10px');
        expect(tree).toHaveStyleRule('transition', 'all .3s ease');
        expect(tree).toHaveStyleRule('transform', 'scale(1)');
    });

    it('test <ThemePreview /> when render with selected props.', () => {
        const { container } = render(<ThemePreview selected />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('transform', 'scale(1.1)');
    });

    it('test <ThemeSettingPicker /> snapshot.', () => {
        const { container } = render(<ThemeSettingPicker />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemeSettingPicker /> when render it.', () => {
        const { container } = render(<ThemeSettingPicker />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
    });

    it('test <ThemePreviewCardFront /> snapshot.', () => {
        const { container } = render(<ThemePreviewCardFront />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemePreviewCardFront /> when render with default props.', () => {
        const { container } = render(<ThemePreviewCardFront />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('border-radius', '10px');
        expect(tree).toHaveStyleRule('color', '#5AC9E8');
        expect(tree).toHaveStyleRule('background-color', '#ffffff');
        expect(tree).toHaveStyleRule('background-image', 'url(card-1.gif)');
        expect(tree).toHaveStyleRule('background-size', 'contain');
        expect(tree).toHaveStyleRule('background-repeat', 'no-repeat');
        expect(tree).toHaveStyleRule('background-position', 'bottom');
    });

    it('test <ThemePreviewCardFront /> when render fontColor and bgColor props.', () => {
        const { container } = render(<ThemePreviewCardFront fontColor="#000" bgColor="#000" />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('color', '#000');
        expect(tree).toHaveStyleRule('background-color', '#000');
    });

    it('test <ThemePreviewCardFrontInner /> snapshot.', () => {
        const { container } = render(<ThemePreviewCardFrontInner />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemePreviewCardFrontInner /> when render it.', () => {
        const { container } = render(<ThemePreviewCardFrontInner />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', '100%');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'flex-start');
        expect(tree).toHaveStyleRule('padding-top', '17%');
        expect(tree).toHaveStyleRule('font-size', '1.2rem');
    });

    it('test <ThemePreviewCardBack /> snapshot.', () => {
        const { container } = render(<ThemePreviewCardBack />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemePreviewCardBack /> when render with default props.', () => {
        const { container } = render(<ThemePreviewCardBack />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', '100%');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('border-radius', '10px');
        expect(tree).toHaveStyleRule('background-color', '#ffffff');
    });

    it('test <ThemePreviewCardBack /> when render with borderColor props.', () => {
        const { container } = render(<ThemePreviewCardBack borderColor="#000" />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('background-color', '#000');
    });

    it('test <ThemePreviewCardBackInner /> snapshot.', () => {
        const { container } = render(<ThemePreviewCardBackInner />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemePreviewCardBackInner /> when render with default props.', () => {
        const { container } = render(<ThemePreviewCardBackInner />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '90%');
        expect(tree).toHaveStyleRule('height', '94%');
        expect(tree).toHaveStyleRule('border-radius', '10px');
        expect(tree).toHaveStyleRule('color', '#ffffff');
        expect(tree).toHaveStyleRule('background-color', '#5bb1d4');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('background-repeat', 'no-repeat');
        expect(tree).toHaveStyleRule('background-position', 'center 40%');
        expect(tree).toHaveStyleRule('background-image', 'url(capoo-cat-logo.png)');
        expect(tree).toHaveStyleRule('background-size', 'contain');
        expect(tree).toHaveStyleRule('padding-top', '45%');
    });

    it('test <ThemePreviewCardBackInner /> when render with fontColor and bgColor props.', () => {
        const { container } = render(<ThemePreviewCardBackInner fontColor="#000" bgColor="#000" />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('color', '#000');
        expect(tree).toHaveStyleRule('background-color', '#000');
    });

    it('test <ThemeSetting /> snapshot.', () => {
        const { container } = render(<ThemeSetting />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemeSetting /> when render it.', () => {
        const { container } = render(<ThemeSetting />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('align-content', 'space-around');
        expect(tree).toHaveStyleRule('flex-wrap', 'wrap');
        expect(tree).toHaveStyleRule('margin-top', '15px');
    });

    it('test <ThemeSettingList /> snapshot.', () => {
        const { container } = render(<ThemeSettingList />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemeSettingList /> when render it.', () => {
        const { container } = render(<ThemeSettingList />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('margin-bottom', '15px');
    });

    it('test <ThemeSettingItem /> snapshot.', () => {
        const { container } = render(<ThemeSettingItem />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ThemeSettingItem /> when render with default props.', () => {
        const { container } = render(<ThemeSettingItem />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('position', 'relative');
        expect(tree).toHaveStyleRule('display', 'inline-block');
        expect(tree).toHaveStyleRule('border-radius', '5px');
        expect(tree).toHaveStyleRule('padding', '2px 7px');
        expect(tree).toHaveStyleRule('margin-right', '10px');
        expect(tree).toHaveStyleRule('transition', 'transform .3s ease');
        expect(tree).toHaveStyleRule('background-color', '#ffffff');
        expect(tree).toHaveStyleRule('color', '#5bb1d4');
        expect(tree).toHaveStyleRule('transform', 'scale(1)');
        expect(tree).toHaveStyleRule('content', 'attr(data-title)', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('margin-right', '0', {
            modifier: ':last-child'
        });
    });

    it('test <ThemeSettingItem /> when render with selected props.', () => {
        const { container } = render(<ThemeSettingItem selected />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('background-color', '#5bb1d4');
        expect(tree).toHaveStyleRule('color', '#ffffff');
        expect(tree).toHaveStyleRule('transform', 'scale(1.1)');
    });
});