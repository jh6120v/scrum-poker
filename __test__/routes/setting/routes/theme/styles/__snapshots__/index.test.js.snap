// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test theme styles. test <ThemePreview /> snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100px;
  max-width: 22%;
  margin: 6.5px 0;
  border-radius: 10px;
  -webkit-transition: all .3s ease;
  transition: all .3s ease;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0::after {
  content: '';
  display: block;
  padding-bottom: 140%;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemePreviewCardBack /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  background-color: #ffffff;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemePreviewCardBackInner /> snapshot. 1`] = `
.c0 {
  width: 90%;
  height: 94%;
  border-radius: 10px;
  color: #ffffff;
  background-color: #5bb1d4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-repeat: no-repeat;
  background-position: center 40%;
  background-image: url(capoo-cat-logo.png);
  background-size: contain;
  padding-top: 45%;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemePreviewCardFront /> snapshot. 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 10px;
  color: #5AC9E8;
  background-color: #ffffff;
  background-image: url(card-1.gif);
  background-size: contain;
  background-repeat: no-repeat;
  background-position: bottom;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemePreviewCardFrontInner /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding-top: 17%;
  font-size: 1.2rem;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemeSetting /> snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 15px;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemeSettingItem /> snapshot. 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  border-radius: 5px;
  padding: 2px 7px;
  margin-right: 10px;
  -webkit-transition: -webkit-transform .3s ease;
  -webkit-transition: transform .3s ease;
  transition: transform .3s ease;
  background-color: #ffffff;
  color: #5bb1d4;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0:before {
  content: attr(data-title);
}

.c0:last-child {
  margin-right: 0;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemeSettingList /> snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 15px;
}

<div
  class="c0"
/>
`;

exports[`test theme styles. test <ThemeSettingPicker /> snapshot. 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
/>
`;
